docs/DeveloperGuide.md

```diff
@@ -1,29 +1,156 @@
+
+Software running flow:
+
+Upon start, Recipeditor will check load or create saves.
+
+During software run, it will repeat iterations of reading and executing commands.
+In each iteration,raw user inputs are read from CLI. They will be
+interpreted into commands for the software to execute.
+Each execution will either write or read the list of recipes
+depending on the command. Finally, the result of current iteration
+is reflected to the user.
+
+Termination of software purges all temporary data, while saved changes
+can be loaded from saves upon next software launch.
+
+
+### Ui Component
+The UI component is responsible for all user interfaces of the application.
+
+<p align="center" width="100%">
+  <img width="80%" src="images/UiClassDiagram.png" alt="Ui Class Diagram"/>
+</p>
+
+**API:** `Ui.java`
+1. `Ui` takes `CommandResult` as a parameter to show the output message after a command is completed.
+2. `AddMode` calls `Recipe` to add new recipe into the list.
+3. `AddMode` calls `Ingredient` to parse ingredients according to its name, amount and unit.
+
+### Storage Component
+The storage component allows data to be read from and saved to a storage file.
+
+<p align="center" width="100%">
+  <img width="80%" src="images/StorageClassDiagram.png" alt="Storage Class Diagram"/>
+</p>
+
+**API:** `Storage.java`
+1. `Storage` calls `Recipe` when saving data from `RecipeList` to an external storage file.
+2. `Storage` calls `RecipeList` when loading recipe data from external storage file to.
+3. `Storage` calls `Ui` to show relevant messages to the user.
+
+The external storage file contains:
+- Recipe Name
+- Recipe Description
+- Recipe Ingredients (name, amount, unit)
+- Recipe Steps
+
+### Command Component
+The command component features a list of commands falls under `Command`,
+identified from user input for the software to carry out certain tasks.
+A `CommandResult` is returned from `execute()` method call of each `Command`.
+The `CommandResult` consists of a single error message in `String`.
+
+<p align="center" width="100%">
+  <img width="80%" src="images/CommandClassDiagram.png" alt="Storage Class Diagram"/>
+</p>
+
+Each subclass of `Command` has their own attributes and `CommandResult`
+from `Execute` method, allowing them to perform respective tasks.
+
+All types of`Command`and their functionalities are explained below:
+
+`AddCommand`: Add a valid `Recipe` to `RecipeList`, otherwise shows error message
+for `invalid Recipe`
+
+`DeleteCommand`: Remove an existing `Recipe` at a valid index from `RecipeList`,
+otherwise show error message on `index out of bound`
+
+`ExitCommand`: Deliver a `CommandResult` to terminate software run.
+
+`InvalidCommand`: Deliver a `CommandResult` of invalid command
+
+`ListCommand`: Print all formatted `Recipe` in `RecipeList` to screen
+
+`ViewCommand`: View an existing `Recipe` at a valid index from `RecipeList`,
+otherwise show error message on `index out of bound`
```

Comment:
Sequence diagrams can be used here to better visualize the flow of how the different commands work. Perhaps elaborate more on how the AddCommand method checks if the Recipe is valid